images/cfw/opt/x1plus/lib/python/x1plus/services/x1plusd/handler.py

```diff
@@ -0,0 +1,66 @@
+import os
+import json
+import logging
+import subprocess
+
+logger = logging.getLogger(__name__)
+
+# Handles callbacks and topic handlers for X1Plus MQTT topics
+# Note - this is just a starting point, quite a bad one, so I plan on rewriting this
+#
+# custom topics and handlers so far:
+# 1) x1plus/settings: x1plus settings set/get
+# 2) x1plus/update: x1plus ota check_now(), etc
+
+class MQTTHandlers:
+    def __init__(self, publisher):
+        self.publisher = publisher
+        
+    async def x1plus_default(self, message):
+        topic = message.topic.value
+        logger.info(f"Received message on {topic} - {message.payload.decode('utf-8')}")
+    
+    # this is pure dogshit and I am going to delete it as soon as it's clear how we want to set this up
+    async def x1plus_update(self, message):
+        topic = message.topic.value
+        logger.info(f"Received update message on {topic}")
+        try:
+            cmd = message.payload.decode('utf-8').strip().split(" ")
+            resp = self.subproc(["x1plus", "ota", cmd[0]] + cmd[1:])
+            await self.publisher("x1plus/update", resp)
+        except Exception as e:
+            logger.error(f"Error handling update command: {e}")
```

Comment:
This I really like, and is something that I kind of hadn't considered.  It sort of seems to me like there are actually three MQTT-related things that are probably separate arcs of work:

* Getting data that normally is broadcast to the MQTT (or DDS) bus into x1plus client apps on the system;
* Having x1plus client apps on the system make requests over the system MQTT (or DDS) bus; and,
* Getting x1plus internal state (OTA info, settings changes, etc.) connected so that other MQTT users (like HA) can get access to them.

I'm wondering if, as gross as it is, we should continue to use the standard Bambu MQTT approach -- broadcast our announcements on `device/{sn}/response` and take commands on `device/{sn}/request`.  If nothing else, MQTT users are used to that at this point.  We could publish them with our own key, `"x1plus_ota": { "command": "status" }`, or whatever, though.